### PR TITLE
Fix and add more benchmarks for Micronaut

### PR DESCRIPTION
**Pull request by dstepanov**
This PR:
- Updates Micronaut to the latest version
- Corrects the bad performance because of the small pool size
- Add different ways to access DB: JDBC, R2DBC
- Add benchmarks with MongoDB

### CODE REVIEW

**Comment by graemerocher** [comment]
See discussion here https://github.com/TechEmpower/FrameworkBenchmarks/discussions/7618

**Comment by dstepanov** [comment]
@nbrady-techempower Any idea why did it fail?

**Comment by dstepanov** [comment]
I see it failed on:
```
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of io.vertx.core.logging.JULLogDelegateFactory are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=io.vertx.core.logging.JULLogDelegateFactory.
```
Will fix it soon.

**Comment by dstepanov** [comment]
@nbrady-techempower Can you please run the build?

**Comment by dstepanov** [comment]
@nbrady-techempower ping

**Comment by dstepanov** [comment]
@nbrady-techempower The update fails with the following:
```
     Only 10108 rows updated in the database out of roughly 10240 expected.
```
I think it's because the random didn't select all the possible rows. Is it possible? The random selection is shared with other subprojects that did pass.

**Comment by dstepanov** [comment]
I have removed the failing projects for now

**Comment by dstepanov** [comment]
@pfalls-techempower @nbrady-techempower Can you please rerun the build?

**Comment by dstepanov** [comment]
Looks like it passed! 🎉

**Review by devgkz — APPROVED** [review]

File: frameworks/Java/micronaut/common-test/src/main/resources/fortunes.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    <tr><td>2</td><td>message1</td></tr>
+    <tr><td>3</td><td>message2</td></tr>
+    <tr><td>4</td><td>message3</td></tr>
+    <tr><td>5</td><td>message4</td></tr>
+    <tr><td>6</td><td>message5</td></tr>
+    <tr><td>7</td><td>message6</td></tr>
+    <tr><td>8</td><td>message7</td></tr>
+    <tr><td>9</td><td>message8</td></tr>
+    <tr><td>10</td><td>message9</td></tr>
+</table></body></html>
```
Comment (nbrady-techempower) [inline frameworks/Java/micronaut/common-test/src/main/resources/fortunes.html:13]: This template looks like it assumes that there will be exactly 10 fortunes. Fortune templates need to be ready for any size list. Future verifications will test this by changing the size of the fortunes.

File: frameworks/Java/micronaut/common/src/main/java/benchmark/controller/AbstractBenchmarkController.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class AbstractBenchmarkController {
+
+    protected final Integer[] boxed = IntStream.range(1, 10001).boxed().toArray(Integer[]::new);
+
+    protected List<Fortune> createFortunes() {
+        List<Integer> fortuneMessages = IntStream.range(0, 10).boxed().collect(Collectors.toList());
+        List<Fortune> fortunes = new ArrayList<>(fortuneMessages.size());
```
Comment (nbrady-techempower) [inline frameworks/Java/micronaut/common/src/main/java/benchmark/controller/AbstractBenchmarkController.java:19]: Same note here. This should be dynamically sized.

**Review by nbrady-techempower — CHANGES_REQUESTED** [review]
Looks like multiple places in code violate [Rule 5 of the Fortune Test Requirements](https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Project-Information-Framework-Tests-Overview#fortunes)

> The list data structure must be a dynamic-size or equivalent and should not be dimensioned using foreknowledge of the row-count of the database table.

File: frameworks/Java/micronaut/common-test/src/main/resources/fortunes.html (hunk 1 above)
Comment (graemerocher) [reply frameworks/Java/micronaut/common-test/src/main/resources/fortunes.html:13]: I don't think the template is actually used the one that is used is https://github.com/TechEmpower/FrameworkBenchmarks/blob/c7cbcf9723aa6c48e6b784a5a6631efc1b48b381/frameworks/Java/micronaut/common/src/main/resources/views/fortunes.rocker.html

@dstepanov can you delete this one?

**Comment by dstepanov** [comment]
@nbrady-techempower All your comments are related to the code that is used for the internal unit testing which is not a requirement .

File: frameworks/Java/micronaut/common/src/main/java/benchmark/controller/AbstractBenchmarkController.java (hunk 2 above)
Comment (dstepanov) [reply frameworks/Java/micronaut/common/src/main/java/benchmark/controller/AbstractBenchmarkController.java:19]: This creates the test data for the unit testing

File: frameworks/Java/micronaut/common-test/src/main/resources/fortunes.html (hunk 1 above)
Comment (dstepanov) [reply frameworks/Java/micronaut/common-test/src/main/resources/fortunes.html:13]: No, it’s used for the unit testing validation

**Comment by dstepanov** [comment]
@nbrady-techempower  Can this be merged?

**Pull request merged**